src/boot/bootctl.c

```diff
@@ -1811,7 +1823,44 @@ static int verb_list(int argc, char *argv[], void *userdata) {
         else
                 (void) boot_entries_augment_from_loader(&config, efi_entries, /* only_auto= */ false);
 
-        if (config.n_entries == 0)
+        if (!FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF)) {
+
+                pager_open(arg_pager_flags);
+
+                for (size_t i = 0; i < config.n_entries; i++) {
+                        _cleanup_free_ char *opts = NULL;
+                        BootEntry *e = config.entries + i;
+                        _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+
+                        if (!strv_isempty(e->options)) {
+                                opts = strv_join(e->options, " ");
+                                if (!opts)
+                                        return log_oom();
+                        }
+
+                        r = json_build(&v, JSON_BUILD_OBJECT(
+                                                       JSON_BUILD_PAIR_CONDITION(e->id, "id", JSON_BUILD_STRING(e->id)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->path, "path", JSON_BUILD_STRING(e->path)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->root, "root", JSON_BUILD_STRING(e->root)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->title, "title", JSON_BUILD_STRING(e->title)),
+                                                       JSON_BUILD_PAIR_CONDITION(boot_entry_title(e), "showTitle", JSON_BUILD_STRING(boot_entry_title(e))),
+                                                       JSON_BUILD_PAIR_CONDITION(e->sort_key, "sortKey", JSON_BUILD_STRING(e->sort_key)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->version, "version", JSON_BUILD_STRING(e->version)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->machine_id, "machineId", JSON_BUILD_STRING(e->machine_id)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->architecture, "architecture", JSON_BUILD_STRING(e->architecture)),
+                                                       JSON_BUILD_PAIR_CONDITION(opts, "options", JSON_BUILD_STRING(opts)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->kernel, "linux", JSON_BUILD_STRING(e->kernel)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->efi, "efi", JSON_BUILD_STRING(e->efi)),
+                                                       JSON_BUILD_PAIR_CONDITION(!strv_isempty(e->initrd), "initrd", JSON_BUILD_STRV(e->initrd)),
+                                                       JSON_BUILD_PAIR_CONDITION(e->device_tree, "deviceTree", JSON_BUILD_STRING(e->device_tree)),
+                                                       JSON_BUILD_PAIR_CONDITION(!strv_isempty(e->device_tree_overlay), "deviceTreeOverlay", JSON_BUILD_STRV(e->device_tree_overlay))));
```

Comment:
sd-boot only does one device tree (the last one wins). So this being a strv is wrong…

Reply:
Also, afaik it doesn't do overlays. So calling it one is also wrong…

Reply:
The field is called `devicetree-overlay`, so this should be "devicetreeOveraly".
Also "devicetree" above. Let's maintain a consistent transliteration from the keys in the spec. Our internal variable names are less important.

I think making this a strv is OK: we may add support for multiple in the future, and then the format won't be broken.

Reply:
well, the spec and the parse know the overlays, so we should show them here. If sd-boot currently supports them is another thing. In theory at least sd-boot is *one* implementation of the boot loader spec, but the spec should be generic.